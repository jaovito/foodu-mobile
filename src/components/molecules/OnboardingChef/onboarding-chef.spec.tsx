import {fireEvent, render} from '@test-utils';
import {OnboardingChef} from '.';

describe('OnboardingChef Tests', () => {
  const title = 'Title';
  const description = 'Description';
  const onPress = jest.fn();

  it('should render correctly', () => {
    const onboarding = render(
      <OnboardingChef
        title={title}
        description={description}
        onPress={onPress}
      />,
    );

    expect(onboarding).toBeDefined();
  });

  it('should be able to find the title of the onboarding container', async () => {
    const {findByText} = render(
      <OnboardingChef
        title={title}
        description={description}
        onPress={onPress}
      />,
    );

    const text = await findByText(title);

    expect(text).toBeTruthy();
  });

  it('should be able to find the description of the onboarding container', async () => {
    const {findByText} = render(
      <OnboardingChef
        title={title}
        description={description}
        onPress={onPress}
      />,
    );

    const text = await findByText(description);

    expect(text).toBeTruthy();
  });

  it('should be able to call onPress function', () => {
    const onboarding = render(
      <OnboardingChef
        title={title}
        description={description}
        onPress={onPress}
      />,
    );

    fireEvent.press(onboarding.getByTestId('ArrowButtonID'));

    expect(onPress).toBeCalledTimes(1);
  });

  it('should match with snapshot', () => {
    const onboarding = render(
      <OnboardingChef
        title={title}
        description={description}
        onPress={onPress}
      />,
    ).toJSON();

    expect(onboarding).toMatchSnapshot();
  });
});
