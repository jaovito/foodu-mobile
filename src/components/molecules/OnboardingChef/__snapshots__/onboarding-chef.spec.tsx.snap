// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnboardingChef Tests should match with snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    [
      {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "flex-end",
        "width": "100%",
      },
    ]
  }
>
  <Image
    source={
      {
        "testUri": "../../../src/assets/chef.png",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#FE8C00",
          "borderRadius": 59,
          "justifyContent": "center",
          "marginBottom": 61,
          "marginLeft": 80,
          "marginRight": 80,
          "marginTop": 0,
          "paddingBottom": 39,
          "paddingLeft": 39,
          "paddingRight": 39,
          "paddingTop": 39,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#0D0D0D",
            "fontFamily": "Poppins_400Regular",
            "fontSize": 31,
          },
          {
            "color": "#FFFFFF",
            "fontFamily": "Poppins_600SemiBold",
            "fontSize": 63,
            "lineHeight": 75,
            "textAlign": "center",
          },
        ]
      }
    >
      Title
    </Text>
    <Text
      style={
        [
          {
            "color": "#0D0D0D",
            "fontFamily": "Poppins_400Regular",
            "fontSize": 31,
          },
          {
            "color": "#FFFFFF",
            "fontFamily": "Poppins_400Regular",
            "fontSize": 27,
            "lineHeight": 41,
            "marginBottom": 69,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 20,
            "textAlign": "center",
          },
        ]
      }
    >
      Description
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 92,
          "height": 184,
          "justifyContent": "center",
          "opacity": 1,
          "width": 184,
        }
      }
      testID="ArrowButtonID"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": undefined,
              "fontSize": 12,
            },
            [
              {
                "color": "#FE8C00",
                "fontSize": 71,
              },
            ],
            {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        ï„’
      </Text>
    </View>
  </View>
</View>
`;
